examples/audio/soundtrack.rs

```diff
@@ -0,0 +1,154 @@
+//! This example illustrates how to load and play different soundtracks
+
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (cycle_game_state, fade_in, fade_out))
+        .add_systems(Update, change_track.run_if(resource_changed::<GameState>))
+        .run();
+}
+
+// This resource simulates game states
+#[derive(Resource, Default)]
+enum GameState {
+    #[default]
+    Peaceful,
+    Battle,
+}
+
+// This timer simulates game state changes
+#[derive(Resource)]
+struct GameStateTimer(Timer);
+
+//  This resource will hold the track list for your soundtrack
+#[derive(Resource)]
+struct SoundtrackPlayer {
+    track_list: Vec<Handle<AudioSource>>,
+}
+
+impl SoundtrackPlayer {
+    fn new(track_list: Vec<Handle<AudioSource>>) -> Self {
+        Self {
+            track_list
+        }
+    }
+}
+
+// This component will be attached to an entity to fade the audio in
+#[derive(Component)]
+struct FadeIn;
+
+// This component will be attached to an entity to fade the audio out
+#[derive(Component)]
+struct FadeOut;
+
+fn setup(asset_server: Res<AssetServer>, mut commands: Commands) {
+    // Instantiate the game state resources
+    commands.insert_resource(GameState::default());
+    commands.insert_resource(GameStateTimer(Timer::from_seconds(10.0, TimerMode::Repeating)));
+
+    // Create the track list
+    let track_1 = asset_server.load::<AudioSource>("sounds/Mysterious accoustic guitar.ogg");
```

Comment:
Yeah, the way this propagates through is unfortunate.

Reply:
What's the best approach for me to fix this? Do I make a suggestion for every line of code that is affected, or do I "Request Changes" ? Still kind of new to contributing to projects, so unsure of how to proceed from here.

Reply:
As the author, just push changes to your branch that fix the issue.